rpc/grpcbuild/grpcbuild.go

```diff
@@ -103,12 +103,21 @@ func (c *grpcClient) SetPhase(phase rpc.Phase, pmd *rpc.PullMetadata) error {
 	c.currentPhase = phase
 	c.phaseSequenceNum += 1
 
+	statusData := &pb.SetPhaseRequest_PullMetadata{}
+	if pmd != nil {
```

Comment:
If `pmd = nil` should it be logged and immediately aborted -- possibly w/ a callback to report a failure?

Reply:
No. Not every phase update has extra metadata to send. Even if it is nil, the buildmanager will handle checking if it's set or not.

Reply:
These are just appended to the phase change log entry and are not critical.